File: apps/ios/ScribbleFit/LocalPackages/Core/CoreCommon/Sources/CoreCommon/Calculations.swift
import Foundation

public enum Calculations {
    /**
     * Epley formula: Weight × (1 + Reps/30)
     */
    public static func calculate1RM(weight: Float, reps: Int) -> Float {
        if reps <= 0 { return 0 }
        if reps == 1 { return weight }
        return weight * (1.0 + Float(reps) / 30.0)
    }

    public static func calculateVolume(weight: Float?, reps: Int) -> Float {
        return (weight ?? 0) * Float(reps)
    }
}

File: apps/android/core/common/src/main/java/com/scribblefit/core/common/Calculations.kt
package com.scribblefit.core.common

object Calculations {
    /**
     * Epley formula: Weight × (1 + Reps/30)
     */
    fun calculate1RM(weight: Float, reps: Int): Float {
        if (reps <= 0) return 0f
        if (reps == 1) return weight
        return weight * (1f + reps.toFloat() / 30f)
    }

    fun calculateVolume(weight: Float?, reps: Int): Float {
        return (weight ?: 0f) * reps
    }
}
